server/user_cloud.go

```diff
@@ -71,10 +71,20 @@ func httpACUserInteractive(jci *jiraCloudInstance, w http.ResponseWriter, r *htt
 			Name: username,
 		},
 	}
-	mattermostUserId, secret, err := jci.Plugin.ParseAuthToken(mmToken)
+	mattermostUserId := r.Header.Get("Mattermost-User-ID")
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	requestedUserId, secret, err := jci.Plugin.ParseAuthToken(mmToken)
 	if err != nil {
 		return http.StatusUnauthorized, err
 	}
+
+	if mattermostUserId != requestedUserId {
```

Comment:
Checks to make sure the user starting the flow is the one finishing it. (otherwise the link is portable to unauthorized browsers) Not a huge deal, but nice to prevent.

Reply:
Well, I gave it thumbs up, then remembered why I didn't do it in the first place, tested it and I doesn't work on my server.

 On my server (with ngrok) I've never seen `Mattermost-User-ID` come through for these requests, always blank. I've been assuming that we rely on explicit request params/headers rather than cookies, and that running the iframe inside Jira somehow messed it up. Probably worth investigating - it'd simplify the flow a lot if we could MM-authenticate these requests as you proposed.

Reply:
I wonder if it's stripped by the server, just like the Authorization header? And if so, can we maybe change that in server 5.14 (for the future)?

If we don't get `Mattermost-User-ID`, can we really secure this from the portability attack you mentioned? Is that a serious problem?

Reply:
This works for me though. It should because the request back to the MM server should include the authentication cookie.

Reply:
How do you expose your server to Jira, direct NAT?

Reply:
Using a server on AWS.
@levb @cpoile Can ether of you reproduce the problem with these changes?

Reply:
I just tried it. Disconnected and reconnected to the cloud mmtest instance. No problems at all through my ngrok. Works!

Reply:
Tested some more. It works for me in Safari, does not in Chrome. There is no cookie coming in the request from Chrome. Probably has something to do with my browser settings, which I don't think I customized much. @cpoile @crspeller

Reply:
Had "Block Third Party Cookies" turned on. Turning it off makes the use-case work. I am not sure what the default in Chrome is, looking.

Reply:
Confirmed that Chrome does _not_ block by default, so I guess it's ok.

We should maybe add copy to the error page that would recommend checking the browser settings? @jasonblais?

Reply:
@levb @crspeller So if someone blocks third party cookies on Chrome, they won't be able to successfully connect their account to Jira? Is that correct?

That seems like a potentially bad experience? Do we have no way of avoiding that?

Reply:
Filed https://mattermost.atlassian.net/browse/MM-15968 to extend (and style?) the error page/message.